src/components/cylc/tree/index.js

```diff
@@ -14,58 +14,542 @@
+  }
+  if (leftObject.type > rightObject.type) {
+    return 1
+  }
+  // name
+  return DEFAULT_COMPARATOR(leftValue, rightValue) > 0
+}
+
+/**
+ * Declare function used in sortedIndexBy for creating the iteratee.
+ *
+ * @callback SortedIndexByIteratee
+ * @param {Object} value - any object
+ * @returns {string}
+ */
+
+/**
+ * Given a list of elements, and a value to be added to the list, we
+ * perform a simple binary search of the list to determine the next
+ * index where the value can be inserted, so that the list remains
+ * sorted.
  *
- * Every node has data, and a .name property used to display the node in the tree in the UI.
+ * This function uses localeCompare, which will respect the numeric
+ * collation.
  *
- * @param tree {null|CylcTree} - A hierarchical tree
- * @param workflow {null|Object} - GraphQL workflow object
- * @throws {Error} - If the workflow or tree are either null or invalid (e.g. missing data)
- */
-function populateTreeFromGraphQLData (tree, workflow) {
-  if (!tree || !workflow || !containsTreeData(workflow)) {
-    // throw new Error('You must provide valid data to populate the tree!')
-    // a stopped workflow is valid, but won't have anything that we can use
-    // to populate the tree, only workflow data and empty families
+ * This is a simplified version of lodash's function with the same
+ * name, but that respects natural order for numbers, i.e. [1, 2, 10].
+ * Not [1, 10, 2].
+ *
+ * @private
+ * @param array {Array<object>} - list of string values, or of objects with string values
+ * @param value {object} - a value to be inserted in the list, or an object wrapping the value (see iteratee)
+ * @param iteratee {SortedIndexByIteratee=} - an optional function used to return the value of the element of the list}
+ * @param comparator {SortedIndexByComparator=} - function used to compare the newValue with otherValues in the list
+ */
+function sortedIndexBy (array, value, iteratee, comparator) {
+  if (array.length === 0) {
+    return 0
+  }
+  // If given a function, use it. Otherwise, simply use identity function.
+  const iterateeFunction = iteratee || ((value) => value)
+  // If given a function, use it. Otherwise, simply use locale sort with numeric enabled
+  const comparatorFunction = comparator || ((leftObject, leftValue, rightObject, rightValue) => DEFAULT_COMPARATOR(leftValue, rightValue) > 0)
+  let low = 0
+  let high = array.length
+
+  const newValue = iterateeFunction(value)
+
+  while (low < high) {
+    const mid = Math.floor((low + high) / 2)
+    const midValue = iterateeFunction(array[mid])
+    const higher = comparatorFunction(value, newValue, array[mid], midValue)
+    if (higher) {
+      low = mid + 1
+    } else {
+      high = mid
+    }
+  }
+  return high
+}
+
+/**
+ * @param {Workflow} workflow
+ */
+function clear (workflow) {
+  ['tree', 'lookup'].forEach(each => {
+    Object.keys(workflow[each]).forEach(key => {
+      Vue.delete(workflow[each], key)
+    })
+  })
+}
```

Comment:
Lots of changes in this file. But you can mostly ignore them. The summary of what was changed is the following:

- No more `CylcTree` object that hold a `tree` structure and a `lookup`
- The functions of the `CylcTree` object were re-wrote into functions that accept the same parameters as before, plus a `workflow` parameter
- The `workflow` parameter comes from Vuex store; it has the following structure: `{ tree: {}, lookup: {}  }`

The `workflow.lookup` contains all the **nodes** of the tree in a flat hashmap. The `workflow.tree` contains the same data, but following a hierarchy. The `.node` of each entry in both `.lookup` and `.tree` points to the `workflows/lookup` global state (there are two lookups, maybe will rename one later).